mtags/src/main/scala/scala/meta/internal/pc/InlineValueProvider.scala

```diff
@@ -0,0 +1,148 @@
+        if (d.termNameRange.encloses(refProvider.position)) inlineAll(d, refs)
+        else inlineOne(d, refs)
+      }
+      .getOrElse(Left(InlineValueProvider.Errors.didNotFindDefinition))
+  }
+
+  private def inlineAll(
+      definition: Definition,
+      references: List[Reference]
+  ): Either[String, List[l.TextEdit]] =
+    if (!definition.isLocal) Left(InlineValueProvider.Errors.notLocal)
+    else
+      Right(
+        definitionTextEdit(definition) :: (references.map(
+          referenceTextEdit(definition)
+        ))
+      )
+
+  private def inlineOne(
+      definition: Definition,
+      references: List[Reference]
+  ): Either[String, List[l.TextEdit]] =
+    if (definition.isLocal && references.length == 1)
+      Right(
+        definitionTextEdit(definition) :: (references.map(
+          referenceTextEdit(definition)
+        ))
+      )
+    else {
+      Right(
+        references
+          .find(_.range.encloses(refProvider.position))
+          .map(referenceTextEdit(definition))
+          .toList
+      )
+    }
+
+  private def referenceTextEdit(
+      definition: Definition
+  )(ref: Reference): l.TextEdit =
+    if (
+      definitionNeedsBrackets(definition.rhs) && referenceNeedsBrackets(
+        ref.parentOffsets
+      )
+    )
+      new l.TextEdit(ref.range, s"""(${definition.rhs})""")
+    else new l.TextEdit(ref.range, definition.rhs)
+
+  private def definitionTextEdit(definition: Definition): l.TextEdit =
+    new l.TextEdit(
+      extendRangeToIncludeWhiteCharsAndTheFollowingNewLine(
+        definition.rangeOffsets.start,
+        definition.rangeOffsets.end,
+        definition.range
+      ),
+      ""
+    )
+
+  private def extendRangeToIncludeWhiteCharsAndTheFollowingNewLine(
+      startOffset: Int,
+      endOffset: Int,
+      range: l.Range
+  ): l.Range = {
+    val source = refProvider.text
+    var startWhiteSpaces = 0
+    if (startOffset != 0) {
+      while (
+        source(startOffset - startWhiteSpaces - 1) == ' ' || source(
+          startOffset - startWhiteSpaces - 1
+        ) == '\t'
+      ) {
+        startWhiteSpaces += 1
+      }
+    }
+    var endWhiteSpaces = 0
+    while (
+      source(endOffset + endWhiteSpaces) == ' ' || source(
+        endOffset + endWhiteSpaces
+      ) == '\t' || source(
+        endOffset + endWhiteSpaces
+      ) == ';'
+    ) {
+      endWhiteSpaces += 1
+    }
+    if (source(endOffset + endWhiteSpaces) == '\n')
+      new l.Range(
+        new l.Position(
+          range.getStart.getLine,
+          range.getStart.getCharacter - startWhiteSpaces
+        ),
+        new l.Position(range.getEnd.getLine + 1, 0)
+      )
+    else
+      new l.Range(
+        new l.Position(
+          range.getStart.getLine,
+          range.getStart.getCharacter - startWhiteSpaces
+        ),
+        new l.Position(
+          range.getEnd.getLine,
+          range.getEnd.getCharacter + endWhiteSpaces
+        )
+      )
+  }
```

Comment:
Something along these lines should work:
```suggestion
    @tailrec
    def expand(step: Int, currentIndex: Int) {
      def isWhiteSpace = source(currentIndex) == ' ' || source(currentIndex) == '\t' 
      if (curretIndex >= 0 && currentIndex < source.size && isWhiteSpace) expand(step, currentIndex + step)
    }     
   val endWithSpace = expand(1, endOffset)
   val startWithSpace = expand(-1, startOffset)
   val startPos =  new l.Position(
          range.getStart.getLine,
          range.getStart.getCharacter -  (startOffset - startWithSpace)
        )
    val endPos = if (source(endWithSpace) == '\n') new l.Position(range.getEnd.getLine + 1, 0)
    else
        new l.Position(
          range.getEnd.getLine,
          range.getEnd.getCharacter +  (endWithSpace - endOffset)
        )
      new l.Range(
        startPos,
        endPos
      )
  }
```